src/Stateless/StateMachine.cs

```diff
@@ -33,21 +44,25 @@ class QueuedTrigger
         /// </summary>
         /// <param name="stateAccessor">A function that will be called to read the current state value.</param>
         /// <param name="stateMutator">An action that will be called to write new state values.</param>
-        public StateMachine(Func<TState> stateAccessor, Action<TState> stateMutator) : this()
+        /// <param name="firingMode">Optional specification of fireing mode.</param>
+        public StateMachine(Func<TState> stateAccessor, Action<TState> stateMutator, FiringMode firingMode = FiringMode.Queued) : this()
```

Comment:
Does this need to be a binary breaking change, or could an additional overload be added? For cases where this package is a dependency-of-a-dependency, the change in constructor signature could cause problems.

Reply:
I could add another constructor, but which mode should the current constructor use?

Would the dependency-of-a-dependency issue really be an issue, since the dependency is always (?) on a specific version?

Reply:
The current constructor should use queued, since it's the existing default, correct?

The problem regarding the dependency is when package A depends on the old version, B depends on the new version, and a user wants to use both A and B. With this change, that scenario is impossible, unfortunately.

Reply:
So what would be the best approach? Keep the original constructor(s), and add another set of constructors With the fireing mode?

Reply:
It's a pretty quick transformation from where the PR is now, I think -

 * The extended one (that accepts firing mode) loses the default value (otherwise overload resolution fails for callers)
 * Another constructor is added without the parameter, which just calls the new, longer constructor, passing in the "queued" default value

If you need a hand I can jump in and edit, just give the word :-) Cheers!

Reply:
I think I understood it, take a look 😁